File: app/src/main/java/com/eftimoff/bakingapp/recipedetails/view/RecipeStepsAdapter.java
package com.eftimoff.bakingapp.recipedetails.view;

import android.support.v7.widget.RecyclerView;

import com.eftimoff.bakingapp.app.models.Recipe;
import com.eftimoff.bakingapp.app.models.Step;

public abstract class RecipeStepsAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    interface Callback {

        void onStepClicked(Step step);

    }

    abstract void setRecipe(Recipe recipe);

    abstract void setCallback(Callback callback);

}

File: app/src/main/java/com/eftimoff/bakingapp/recipedetails/view/RecipeStepsAdapter.kt
package com.eftimoff.bakingapp.recipedetails.view

import android.support.v7.widget.RecyclerView
import com.eftimoff.bakingapp.app.models.Recipe
import com.eftimoff.bakingapp.app.models.Step

abstract class RecipeStepsAdapter : RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    interface Callback {
        fun onStepClicked(step: Step)
    }

    abstract fun setRecipe(recipe: Recipe)

    abstract fun setCallback(callback: Callback)

}